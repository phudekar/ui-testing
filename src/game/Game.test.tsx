import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { StateMock } from '@react-mock/state';
import Game from '.';

describe('Game', () => {
    it('should show board with default board with 10x10 size', async () => {
        const { container, getByText, queryByText } = render(<Game />);

        const blocks = container.querySelectorAll('.block');

        expect(blocks.length).toBe(10 * 10)
        expect(getByText('Minesweeper')).toBeVisible();
        expect(queryByText('Game Over!')).toBeNull();
    })

    it('should show game over message with play again button', () => {
        const { queryByText } = render(
            <StateMock state={{ gameOver: true }} >
                <Game />
            </StateMock>);

        expect(queryByText('Game Over!')).toBeVisible();
        expect(queryByText('Retry')).toBeVisible();
    })

    it('should show congratulations message on game complete', () => {
        const { queryByText } = render(
            <StateMock state={{ gameComplete: true }} >
                <Game />
            </StateMock>);

        expect(queryByText('Congratulations! You flagged all the mines.')).toBeVisible();
        expect(queryByText('Play Again')).toBeVisible();
    })

    it('should clear congratulations message on clicking Play again button', () => {
        const { getByText, queryByText } = render(
            <StateMock state={{ gameComplete: true }} >
                <Game />
            </StateMock>);

        fireEvent.click(getByText('Play Again'));

        expect(queryByText('Congratulations! You flagged all the mines.')).toBeNull();

    })

    it('should clear game over message after clicking play again button', () => {
        const { getByText, queryByText } = render(
            <StateMock state={{ gameOver: true }} >
                <Game />
            </StateMock>);

        fireEvent.click(getByText('Retry'));

        expect(queryByText('Game Over!')).toBeNull();
    })

    it('should render component in default state', () => {
        const { container } = render(
            <StateMock >
                <Game />
            </StateMock>);

        expect(container.firstChild).toMatchSnapshot();
    })

    it('should render component in game over state', () => {
        const { container } = render(
            <StateMock state={{ gameOver: true }}  >
                <Game />
            </StateMock>);

        expect(container.firstChild).toMatchSnapshot();
    })
})
