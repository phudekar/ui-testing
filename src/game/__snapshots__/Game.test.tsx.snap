// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should render component in default state 1`] = `
<div>
  <div
    class="title"
  >
    Minesweeper
  </div>
  <div
    class="board"
    data-testid="board"
  >
    <div
      class="row"
    >
      <div
        class="block"
        data-testid="block-0-0"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-0-1"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-0-2"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-0-3"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-0-4"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-0-5"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-0-6"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-0-7"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-0-8"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-0-9"
      >
        <div
          class="mine"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="block"
        data-testid="block-1-0"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-1-1"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-1-2"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-1-3"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-1-4"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-1-5"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-1-6"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-1-7"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-1-8"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-1-9"
      >
        <div
          class="mine"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="block"
        data-testid="block-2-0"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-2-1"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-2-2"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-2-3"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-2-4"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-2-5"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-2-6"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-2-7"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-2-8"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-2-9"
      >
        <div
          class="mine"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="block"
        data-testid="block-3-0"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-3-1"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-3-2"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-3-3"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-3-4"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-3-5"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-3-6"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-3-7"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-3-8"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-3-9"
      >
        <div
          class="mine"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="block"
        data-testid="block-4-0"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-4-1"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-4-2"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-4-3"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-4-4"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-4-5"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-4-6"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-4-7"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-4-8"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-4-9"
      >
        <div
          class="mine"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="block"
        data-testid="block-5-0"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-5-1"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-5-2"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-5-3"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-5-4"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-5-5"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-5-6"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-5-7"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-5-8"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-5-9"
      >
        <div
          class="mine"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="block"
        data-testid="block-6-0"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-6-1"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-6-2"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-6-3"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-6-4"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-6-5"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-6-6"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-6-7"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-6-8"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-6-9"
      >
        <div
          class="mine"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="block"
        data-testid="block-7-0"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-7-1"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-7-2"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-7-3"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-7-4"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-7-5"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-7-6"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-7-7"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-7-8"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-7-9"
      >
        <div
          class="mine"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="block"
        data-testid="block-8-0"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-8-1"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-8-2"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-8-3"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-8-4"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-8-5"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-8-6"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-8-7"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-8-8"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-8-9"
      >
        <div
          class="mine"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="block"
        data-testid="block-9-0"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-9-1"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-9-2"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-9-3"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-9-4"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-9-5"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-9-6"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-9-7"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-9-8"
      >
        <div
          class="mine"
        />
      </div>
      <div
        class="block"
        data-testid="block-9-9"
      >
        <div
          class="mine"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Game should render component in game over state 1`] = `
<div>
  <div
    class="title"
  >
    Minesweeper
  </div>
  <div
    class="board"
    data-testid="board"
  />
  <div
    class="overlay"
  >
    <div
      class="game-over"
    >
      Game Over!
    </div>
    <button
      class="retry-button"
    >
      Retry
    </button>
  </div>
</div>
`;
